### PR TITLE
fix(@desktop/settings): [re-add] ensure new password is confirmed

### LINKED ISSUE #4672: re-add to master: fix(@desktop/settings): ensure new password is confirmed
7ca3766def467b810fb67b5bdd0df8f955f4adf6

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4672

Original issues
https://github.com/status-im/status-desktop/issues/4375
https://github.com/status-im/status-desktop/issues/4376

### CODE REVIEW

**Review by anastasiyaig — APPROVED** [review]

File: ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml [hunk 1]
```diff
@@ -12,113 +12,163 @@ import StatusQ.Popups 0.1
...
-                    + "Your password protects your keys. You need it to unlock Status and transact."
-            }
+            keepHeight: true
+            placeholderText: ""
+            state: "init"
+
+            onTextChanged: if (text != "" && state != "typing") state = "typing"
+            onStateChanged: if (state == "init") resetInternal()
+
+            states: [
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml:152]: can the states be combined to a common for the whole file and all the 3 properties being affected can be changed from there?

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Hey just somethings, let me know what you think about them?

File: ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml [hunk 2]
```diff
@@ -12,113 +12,163 @@ import StatusQ.Popups 0.1
...
     header.title: qsTr("Change password")
 
-    onOpened: {
-        reset();
-    }
+    onOpened: d.reset()
 
-    property var privacyStore
+    QtObject {
+        id: d
```
[the remark this answers is not in the document]
Comment (jrainville) [reply ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml]: Fixed

File: ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml:148]: Hey I just saw "init", "typing" where common thought they could be merged but off course doesn't need to be right now. approving it for now :)

**Review by Khushboo-dev-cpp — APPROVED** [review]

**Review by saledjenic — APPROVED** [review]

File: ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml (hunk 1 above)
Comment (osmaczko) [reply ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml:148]: Initially, this whole logic was extracted to a separate Component: https://github.com/status-im/status-desktop/blob/83854ddd9be940a2344e6f1ae125f1baeb955d8d/ui/imports/shared/controls/PasswordInput.qml
But it was reverted due to: https://github.com/status-im/status-desktop/pull/4400#discussion_r784170200 which unfortunately results in code duplication

The states must not be combined to a common one, each `Input` should have its own state as they are separated entities.

**Review by osmaczko — APPROVED** [review]

**Pull request merged**